### ISSUE TITLE
Javalib "no future development" needs to be documented in-tree

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2760]
Scala Native works on a "professional/community' development model, where contributions
from community members are encouraged. Professionals guide and are the final merge gatekeepers.

It was decided and implemented there would be no further work in Scala core on
 `java.time` and `java.util.locale`. Discussions are underway  to do something similar
to `java.security`.

I propose to create files README (or README_Java_Time, or wordsmithed)
in `java` and `java.util` to say that `java.foo` is not present by intent and
will have no further development. Please use 3rd party libraries;" (wordsmithed).

I also propose that we agree to that convention for future 'evictions`, such as
`java.security`.  

My thinking is that it is natural for potential contributors to look for 'completeness'
and offer contributions for obviously missing features. I believe that letting such
people know early in the process that the intent is no development be done
preserves happiness.  Waiting until a PR is rejected/not_merged brings
not merit to anyone.   

Agreed, in a better world, potential contributors would ask before investing
time. 

One benefit of this convention would be preserving Institutional Memory
about what has been pruned (and in git, when and by whom).

I am not trying to add burdensome overhead. I am trying to walk the 
path of a potential contributor to identify and remove dissatisfaction points.
As a community, there are all too few of us and we want/need to be
welcoming.

Please advise.

**Comment by ekrich** [comment]
Right at the top of this page could be a statement of what is not planned to be added to the core with potential packges to use. <https://scala-native.readthedocs.io/en/latest/lib/javalib.html>

**Comment by LeeTibbert** [comment]
Agreed & good idea.  I think having the proposed READMEs give that URL (possibly with a section reference)
would minimize duplication and increase the likelihood of descriptions being up to date.  The re-direction
would also allow people to see the broader scope of "not planned".
